cpp/src/io/parquet/page_enc.cu

```diff
@@ -154,14 +153,19 @@ __global__ void __launch_bounds__(block_size) gpuInitPageFragments(PageFragment
     // etc...
     s->start_value_idx      = start_row;
     size_type end_value_idx = start_row + s->frag.num_rows;
-    for (size_type i = 0; i < s->col.nesting_levels; i++) {
-      s->start_value_idx = s->col.nesting_offsets[i][s->start_value_idx];
-      end_value_idx      = s->col.nesting_offsets[i][end_value_idx];
+    if (s->col.parent_column != nullptr) {
+      auto col = *(s->col.parent_column);
+      while (col.type().id() == type_id::LIST) {
+        auto offset_col    = col.child(lists_column_view::offsets_column_index);
+        s->start_value_idx = offset_col.element<size_type>(s->start_value_idx);
+        end_value_idx      = offset_col.element<size_type>(end_value_idx);
```

Comment:
why are these set in every iteration?

Reply:
You mean you want to have a temporary variable like I mentioned [here](https://github.com/rapidsai/cudf/pull/7097#discussion_r566222088)?

Or did you mean to ask why we have this repeated call like :
`x = offset_col[x];`

Reply:
@vuule , do you mean the purpose of this code or the fact that this is writing repeatedly to shared storage?
The purpose is to repeatedly apply index to offsets to get the offsets in the final leaf column. Explained in the preceding comment [here](https://github.com/rapidsai/cudf/blob/90b5da019c33dfcfd1c1685b02d305c8d8e2cb70/cpp/src/io/parquet/page_enc.cu#L149-L153)

Reply:
I was asking about the repeated setting, since I missed where the vars were used. But now that I see my mistake, using a temp var does sound good to avoid unnecessary access to shared memory :)